tests/fixtures.py

```diff
@@ -28,6 +31,32 @@ def load_config():
         raise Exception("KUBECONFIG file not found: {0}".format(kube_config_path))
 
 
+@pytest.fixture(scope='session')
+def set_namespace(request, load_config):
+
+    def _setup_namespace():
+        k8s_namespace = os.getenv('K8S_NAMESPACE', '')
+        if k8s_namespace != '':
+            testconfig['namespace'] = k8s_namespace
+
+        print("Run tests in namespace: {0}".format(testconfig['namespace']))
+        v1 = client.CoreV1Api()
+        namespaces_list = [ns.metadata.name for ns in v1.list_namespace().items]
+        if testconfig['namespace'] in namespaces_list:
+            return
+
+        body = client.V1Namespace()
+        body.metadata = client.V1ObjectMeta(name=testconfig['namespace'])
+        v1.create_namespace(body)
```

Comment:
do we need to wait for namespace to be created ?

Reply:
Not really. It is created right away